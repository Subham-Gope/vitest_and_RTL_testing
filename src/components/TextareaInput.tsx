import React from "react";
import { useState, type ChangeEvent } from "react";

const TextareaInput: React.FC = () => {
  const [text, setText] = useState<string>("");

  const handleChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
    setText(e.target.value);
  };

  return (
    <>
      <div>
        <label htmlFor="text-area">Enter Your Text</label>
        <textarea
          id="text-area"
          placeholder="enter your text"
          value={text}
          onChange={handleChange}
        />
      </div>
      {text && <p>Entered Input: {text}</p>}
    </>
  );
};
export default TextareaInput;
