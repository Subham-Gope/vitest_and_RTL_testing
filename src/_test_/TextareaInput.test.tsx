import { describe, test, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TextareaInput from "../components/TextareaInput";

describe("testing the TextareaInput.tsx file", () => {
  test("should type multi-line text", async () => {
    const { container } = render(<TextareaInput />);
    expect(container).toMatchSnapshot();

    // setting up the the userEvent
    const user = userEvent.setup();

    const input = screen.getByRole("textbox", { name: /enter your/i });
    expect(input).toBeInTheDocument();
    expect(input).toHaveValue("");

    // entering the multiline text
    await user.type(input, "Line1{enter}Line2");
    expect(input).toHaveValue("Line1\nLine2");
  });
});
