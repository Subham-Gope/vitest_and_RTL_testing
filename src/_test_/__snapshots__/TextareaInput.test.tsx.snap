// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing the TextareaInput.tsx file > should type multi-line text 1`] = `
<div>
  <div>
    <label
      for="text-area"
    >
      Enter Your Text
    </label>
    <textarea
      id="text-area"
      placeholder="enter your text"
    />
  </div>
</div>
`;
